### PR TITLE
Playwright: Import TypeScript libraries directly into tests.

### PR DESCRIPTION
**Pull request by worldomonation**
#### Changes proposed in this Pull Request

In PR #51146 I had implemented a workflow that required the compilation of TypeScript files into JavaScript prior to import into the test files.

After some feedback from @Automattic/team-calypso-platform, this PR will switch away from that to the practice of importing TypeScript files directly into JavaScript test files.

- add new module to dependencies: `ts-node` . It is in `devDependencies` because of [this StackOverflow answer](https://stackoverflow.com/a/44249080).
- update `.mocharc.yml` to import `ts-node/register`.

#### Testing instructions

Related to #50693
Parent: #51146
Child: #51285

### CODE REVIEW

File: test/e2e/lib/browser-manager.ts [hunk 1]
```diff
@@ -6,8 +6,9 @@
 /**
  * External dependencies
  */
-import { Browser, BrowserContext, chromium } from 'playwright';
+import { chromium } from 'playwright';
 import config from 'config';
+import type { Browser, BrowserContext } from 'playwright';
```
Comment (worldomonation) [inline test/e2e/lib/browser-manager.ts:11]: This addresses https://github.com/Automattic/wp-calypso/pull/51146#discussion_r601741698.

**Comment by scinos** [comment]
FWIW I think we should do a explicit `tsc` compilation (via `yarn build`) like we do in the rest of the repo, and not use `ts-node`.

What are the reasons to prefer `ts-node`?

**Comment by sarayourfriend** [comment]
> What are the reasons to prefer ts-node?

Not using `ts-node` requires importing generated JavaScript from the `dist` directory... that doesn't really make sense to me, but maybe I'm wrong about that @scinos?

**Comment by scinos** [comment]
> Not using ts-node requires importing generated JavaScript from the dist directory... 

I don't see a huge problem with that... But if we want to solve it, then we can change `tsconfig.json` to not output the files to `dist`, but to the same path where the `.ts` lives. Or migrate the specs to TypeScript as well.

**Comment by worldomonation** [comment]
I am uncertain where this conversation leaves us in terms of direction.

What I had envisioned to implement the migrated Playwright end-to-end suite as follows:
- libraries/pages/etc: TypeScript
- specs: JavaScript

Now if it makes sense to unify to TypeScript to both, I don't mind but it does reduce the attractiveness of writing end-to-end tests for non-TS developers.

I do not have an issue with transpiling TypeScript files to JavaScript then importing those files either.

In any case, I am updating my subsequent child PRs with the assumption that I will continue to implement libraries in TypeScript and import them directly to specs written in JavaScript. If it turns out we want to transpile instead, then it should be fairly easy to update the necessary portions.

**Comment by scinos** [comment]
> re: ts-node

I just don't like the extra layer of complexity it adds. I remember having troubles when trying to debug something with VSCode because the `ts-node` layer was swallowing some CLI flags and making the debugger harder to set up in general. I know my argument is subjective and a bit vague, I'd love to have something more solid, sorry.

But my biggest con against `ts-node` is that I think there are simpler solutions to the problem of importing from `../dist/` (starting with just live with it, but I understand that's subjective).

After thinking about this, I think my preferred solution would be to move the libraries to a separate package `@automattic/calypso-e2e` (or better name), that can be written in TS like many other packages we already have in `./packages`. Then the specs (written in JS or TS, doesn't matter) can import that package: JS importing a TS package is something we do in many places already.

**Comment by worldomonation** [comment]
I did some background reading, and boy is this contentious.

In any typical thread (Reddit, StackOverflow, etc.) there are proponents of either side and in more or less equal numbers, split down the middle.

[Reddit thread 1](https://www.reddit.com/r/typescript/comments/9o1zzp/tsnode_vs_compiling_for_development/)
[Reddit thread 2](https://www.reddit.com/r/typescript/comments/8vkvzy/typescript_with_node_should_i_use_tsnode_or_tsc/)
[Stack Overflow](https://stackoverflow.com/questions/60581617/is-it-a-bad-practise-to-use-ts-node-in-production)

This is rather interesting and something I did not expect. Good learning opportunity for me :)

**Comment by noahtallen** [comment]
>  I think my preferred solution would be to move the libraries to a separate package @automattic/calypso-e2e

This is similar to how Gutenberg does it, though they don't use typescript. I also like the idea of being able to import test helpers like `import { createPost } from '@automattic/calypso-e2e'`

**Comment by worldomonation** [comment]
Thanks for feedback all.

I want to clarify how I should proceed here. 
It is looking like merging this PR will have to wait until I come back at end of April. When I return, I want to be ready to implement the agreed solution.

Personally speaking,I felt that transpiling TypeScript to JavaScript then having the tests import the transpiled files worked well, even if it could lead to issues for some developers (forgetting to compile, etc). I felt the transpile approach will allow me to maintain focus on the Playwright migration + rewriting supporting libraries in TypeScript. 

Would it make sense to leave the packaging of libraries to another project to be handled somewhere down the line? I am unfamiliar with JS packaging in general, so if it's a trivial thing - I apologize.

**Comment by scinos** [comment]
It is totally ok to wait until you come back :)
I can help with moving the TS bits to its own package.

**Comment by scinos** [comment]
I moved the new libs to a separate package `@automattic/calypso-e2e`. Feel free to change its name or its structure as you see fit, my initial package is just a demo :)

How to run in prod:
```
yarn workspace @automattic/calypso-e2e build
cd test/e2e
yarn magellan --config=magellan-playwright.json...
```

How to develop it:
```
# In a terminal
yarn workspace @automattic/calypso-e2e build --watch

# In a separate terminal
cd test/e2e
yarn magellan --config=magellan-playwright.json...
```

**Comment by worldomonation** [comment]
> I moved the new libs to a separate package `@automattic/calypso-e2e`. Feel free to change its name or its structure as you see fit, my initial package is just a demo :)

Thanks, it looks good!

**Comment by worldomonation** [comment]
With the additional step to build the package added to the Playwright TC run, tests there pass.
Other failures appear to be flakey failures from mobile tests. I think this PR is good to merge.

**Pull request merged**